packages/components/src/dropdown-menu/index.js

```diff
@@ -12,34 +12,42 @@ import { DOWN } from '@wordpress/keycodes';
 /**
  * Internal dependencies
  */
+import DropdownMenuSeparator from './separator';
 import IconButton from '../icon-button';
 import Dropdown from '../dropdown';
 import { NavigableMenu } from '../navigable-container';
 
 function DropdownMenu( {
+	contentClassName,
+	children,
+	className,
+	controls,
 	icon = 'menu',
 	label,
+	labelPosition,
 	menuLabel,
-	controls,
-	className,
 	position,
 } ) {
-	if ( ! controls || ! controls.length ) {
+	if ( isEmpty( controls ) && ! isFunction( children ) ) {
 		return null;
 	}
 
 	// Normalize controls to nested array of objects (sets of controls)
-	let controlSets = controls;
-	if ( ! Array.isArray( controlSets[ 0 ] ) ) {
-		controlSets = [ controlSets ];
+	let controlSets;
+	if ( ! isEmpty( controls ) ) {
+		controlSets = controls;
+		if ( ! Array.isArray( controlSets[ 0 ] ) ) {
+			controlSets = [ controlSets ];
+		}
 	}
 
 	return (
 		<Dropdown
 			className={ classnames( 'components-dropdown-menu', className ) }
-			contentClassName="components-dropdown-menu__popover"
+			contentClassName={ classnames( 'components-dropdown-menu__popover', contentClassName ) }
 			position={ position }
 			renderToggle={ ( { isOpen, onToggle } ) => {
+				const buttonLabel = isFunction( label ) ? label( isOpen ) : label;
```

Comment:
A workaround to allow providing different labels depending on the `isOpen` value.

Reply:
This is no longer the case.